std/sync/sv_map.go

```diff
@@ -0,0 +1,62 @@
+package sync
+
+import (
+	"sort"
+
+	spec_svs "github.com/named-data/ndnd/std/ndn/svs/v3"
+)
+
+// Map representation of the state vector.
+type svMap map[uint64][]spec_svs.SeqNoEntry
+
+// Get seq entry for a bootstrap time.
+func (m svMap) get(hash uint64, btime uint64) spec_svs.SeqNoEntry {
+	for _, entry := range m[hash] {
+		if entry.BootstrapTime == btime {
+			return entry
+		}
+	}
+	return spec_svs.SeqNoEntry{
+		BootstrapTime: btime,
+		SeqNo:         0,
+	}
+}
+
+// Set seq entry for a bootstrap time.
+func (m svMap) set(hash uint64, btime uint64, seq uint64) {
+	for i, entry := range m[hash] {
+		if entry.BootstrapTime == btime {
+			m[hash][i].SeqNo = seq
+			return
+		}
+	}
+	m[hash] = append(m[hash], spec_svs.SeqNoEntry{
+		BootstrapTime: btime,
+		SeqNo:         seq,
+	})
+	sort.Slice(m[hash], func(i, j int) bool {
+		return m[hash][i].BootstrapTime < m[hash][j].BootstrapTime
+	})
+}
+
+// Check if a svHashMap is newer than another.
+// If existOnly is true, only check if the other has all entries.
```

Comment:
The name `existOnly` sounds confusing. Maybe we can say `uniqueEntryOnly`.

Reply:
Not sure. If this flag is set then it only checks existence of all entries in the other vector without comparing the seq no.